### PR TITLE
Graphspace

### PR DESCRIPTION
**Pull request by annacalissano**
metric space for unlabelled graphs

### CODE REVIEW

File: geomstats/geometry/graphspace.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        Second graph to align.
+
+        Returns
+        -------
+        permutation : array-like, shape=[...,n]
+            node permutation indexes of the second graph.
+
+        References
+        ----------
+        ..[Volgstain2015] Vogelstein JT, Conroy JM, Lyzinski V, Podrazik LJ,
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:182]: There is a typo in the name Volgstain2015 -> Vogelstein2015

File: geomstats/geometry/graphspace.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+        References
+        ----------
+        ..[Jain2009]  Jain, B., Obermayer, K.
+                  "Structure Spaces." Journal of Machine Learning Research 10.11 (2009).
+                  https://www.jmlr.org/papers/v10/jain09a.html.
+        """
+        if matcher == "FAQ":
+            perm = self.faq_matching(base_graph, graph_to_permute)
+        if matcher == "ID":
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:157]: NIT: if -> elif

File: geomstats/geometry/graphspace.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    Parameters
+    ----------
+    nodes : int
+        Number of graph nodes
+    p : int
+        Dimension of euclidean parameter or label associated to a graph.
+
+    References
+    ----------
+    ..[Calissano2020]  Calissano, A., Feragen, A., Vantini, S.
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:25]: 🎉

File: geomstats/geometry/graphspace.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+              “Graph Space: Geodesic Principal Components for aPopulation of
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:26]: NIT: aPopulation -> a Population

File: geomstats/geometry/graphspace.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+
+class _GraphSpace:
+    r"""Class for the Graph Space.
+
+    Graph Space to analyse populations of labelled and unlabelled graphs.
+    The space is a quotient space obtained by applying the permutation
+    action of nodes to the space of adjacency matrices.
+
+    Points are represented by :math:`nodes \times nodes` adjacency matrices.
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:14]: Adjacency matrices can thus be non-symmetric, thus we are looking at directed weighted graphs? Should we specify this?

File: geomstats/geometry/graphspace.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+        """
+        return self.adjmat.random_point(n_samples=n_samples, bound=bound)
+
+    def permute(self, graph_to_permute, permutation):
+        r"""Permutation action applied to graph observation.
+
+        Parameters
+        ----------
+        graph_to_permute : array-like, shape=[..., n, n]
+            input graphs to be permuted
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:81]: NIT: Input graphs to be permuted. 

(upper case at the first letter and "." at the end)

File: geomstats/geometry/graphspace.py [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+        permutation: array-like, shape=[..., n]
+            node permutations
```
Comment (ninamiolane) [inline geomstats/geometry/graphspace.py:83]: Node permutations.

File: geomstats/geometry/graphspace.py (hunk 3 above)
Comment (annacalissano) [reply geomstats/geometry/graphspace.py:25]: ahahah

File: geomstats/geometry/graphspace.py (hunk 5 above)
Comment (annacalissano) [reply geomstats/geometry/graphspace.py:14]: Graph Space to analyse populations of labelled and unlabelled graphs.
    The space focuses on graphs with scalar euclidean attributes on nodes and edges, 
    with a finite number of nodes and both directed and undirected edges. 
    For undirected graphs, use symmeric adjacency matrices. The space is a quotient 
    space obtained by applying the permutation action of nodes to the space 
    of adjacency matrices.

**Review by ninamiolane — APPROVED** [review]
Ma che fantastico!

**Pull request merged**